packages/@aws-cdk/aws-fsx/lib/lustre-file-system.ts

```diff
@@ -0,0 +1,305 @@
+
+  /**
+   * The preferred time to perform weekly maintenance, in the UTC time zone, 24 hour clock. For example: '2:20:30'.
+   *
+   * @default - no preference
+   */
+  readonly weeklyMaintenanceStartTime?: string;
+}
+
+/**
+ * Properties specific to the Lustre version of the FSx file system.
+ */
+export interface FsxLustreFileSystemProps extends FsxFileSystemProps {
+  /**
+   * Additional configuration for FSx specific to Lustre.
+   */
+  readonly lustreConfiguration: LustreConfiguration;
+
+  /**
+   * The subnet that the file system will be accessible from.
+   */
+  readonly vpcSubnet: ISubnet;
+}
+
+/**
+ * The FSx for Lustre File System implementation of IFsxFileSystem.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-fsx-filesystem.html
+ *
+ * @resource AWS::FSx::FileSystem
+ */
+export class FsxLustreFileSystem extends FsxFileSystemBase {
+
+  /**
+   * Import an existing FSx for Lustre file system from the given properties.
+   */
+  public static fromFsxLustreFileSystemAttributes(scope: Construct, id: string, attrs: FsxFileSystemAttributes): IFsxFileSystem {
+    class Import extends FsxFileSystemBase {
+      public readonly fileSystemId = attrs.fileSystemId;
+      public readonly connections = new Connections({
+        securityGroups: [attrs.securityGroup],
+        defaultPort: Port.tcp(FsxLustreFileSystem.DEFAULT_PORT)
+      });
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The default FSx file system type used by FSx for Lustre.
+   */
+  private static readonly DEFAULT_FILE_SYSTEM_TYPE: string = "LUSTRE";
+
+  /**
+   * The default port the file system listens on.
+   */
+  private static readonly DEFAULT_PORT: number = 988;
+
+  /**
+   * The security groups/rules used to allow network connections to the file system.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * The ID that AWS assigns to the file system.
+   */
+  public readonly fileSystemId: string;
+
+  /**
+   * The encapsulated L1 file system.
+   */
+  private readonly fsxFileSystem: CfnFileSystem;
+
+  constructor(scope: Construct, id: string, props: FsxLustreFileSystemProps) {
+    super(scope, id);
+
+    this.validateProps(props);
+
+    const securityGroup = (props.securityGroup || new SecurityGroup(this, 'FsxLustreSecurityGroup', {
+      vpc: props.vpc
+    }));
+    securityGroup.addIngressRule(Peer.anyIpv4(), Port.tcp(FsxLustreFileSystem.DEFAULT_PORT), 'Default FSx port ingress');
+    securityGroup.addEgressRule(Peer.anyIpv4(), Port.tcp(FsxLustreFileSystem.DEFAULT_PORT), 'Default FSx port egress');
+
+    this.connections = new Connections({
+      securityGroups: Array.of(securityGroup),
+      defaultPort: Port.tcp(FsxLustreFileSystem.DEFAULT_PORT)
+    });
+
+    this.fsxFileSystem = new CfnFileSystem(this, 'FsxLustreFileSystem', {
+      fileSystemType: FsxLustreFileSystem.DEFAULT_FILE_SYSTEM_TYPE,
+      subnetIds: Array.of(props.vpcSubnet.subnetId),
```

Comment:
Yeah so it does take MULTIPLE subnets, no?

When you take a subnetselection (defaulting to `undefined`) you can do:

```
subnetIds: props.vpc.selectSubnets(props.subnetSelection).subnetIds
```

Reply:
It does take multiple subnets, but only a very specific use case for FSx for Windows Server allows them. I tried removing the option to set multiple subnets from the Lustre construct as it is invalid.